### PR TITLE
xdr: expose the source hash of generated xdr

### PR DESCRIPTION
**Pull request by tsachiherman**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ X ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ X ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ X ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ X ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

This PR expose the hash of the source for the generated XDR. 

### Why

This allows derived components to report about the concrete XDR version that the module was compiled against.

### Known limitations

N/A

### CODE REVIEW

File: xdr/main.go [hunk 1]
```diff
@@ -15,6 +16,13 @@ import (
 	"github.com/stellar/go/support/errors"
 )
 
+// CommitHash is the commit hash that was used to generate the xdr in this folder.
+// During the process of updating the XDR, the text file below is being updated.
+// Than, during compile time, the file content are being embedded into the given string.
+//
+//go:embed xdr_commit_generated.txt
+var CommitHash string
```
Comment (leighmcculloch) [inline xdr/main.go:24]: Love the simplicity of this.

File: Makefile [hunk 2]
```diff
@@ -42,6 +42,9 @@ gxdr/xdr_generated.go: $(XDRS)
 xdr/%.x:
 	curl -Lsf -o $@ https://raw.githubusercontent.com/stellar/stellar-xdr-next/$(XDRNEXT_COMMIT)/$(@F)
```
Comment (leighmcculloch) [inline Makefile:43]: Note that the repo is renamed.
```suggestion
	curl -Lsf -o $@ https://raw.githubusercontent.com/stellar/stellar-xdr/$(XDRNEXT_COMMIT)/$(@F)
```

**Review by leighmcculloch — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (tsachiherman) [reply Makefile:43]: yes, I'm well aware of that. However, @paulbellamy PR already addresses exactly that.. so I don't want it in my PR.

File: xdr/main.go [hunk 3]
```diff
@@ -15,6 +16,13 @@ import (
 	"github.com/stellar/go/support/errors"
 )
 
+// CommitHash is the commit hash that was used to generate the xdr in this folder.
+// During the process of updating the XDR, the text file below is being updated.
+// Than, during compile time, the file content are being embedded into the given string.
```
Comment (2opremio) [inline xdr/main.go:21]:
```suggestion
// Then, during compile time, the file content are being embedded into the given string.
```

**Review by 2opremio — APPROVED** [review]

File: xdr/main.go (hunk 3 above)
Comment (tsachiherman) [reply xdr/main.go:21]: will fix. thanks!

**Pull request merged**